extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoClientRecorder.java

```diff
@@ -20,35 +20,20 @@
 
     public Supplier<MongoClientSupport> mongoClientSupportSupplier(List<String> codecProviders, List<String> bsonDiscriminators,
             List<ConnectionPoolListener> connectionPoolListeners, boolean disableSslSupport) {
-        return new Supplier<MongoClientSupport>() {
-            @Override
-            public MongoClientSupport get() {
-                return new MongoClientSupport(codecProviders, bsonDiscriminators, connectionPoolListeners, disableSslSupport);
-            }
-        };
+        return () -> new MongoClientSupport(codecProviders, bsonDiscriminators, connectionPoolListeners, disableSslSupport);
```

Comment:
Thanks for this PR! Please revert all the lambda changes. We don't want lambdas in runtime code as they consume extra memory.

Thanks!

Reply:
Uh, sure. Wasn't aware of that, thanks!